src/OptimalTransport.jl

```diff
@@ -184,15 +194,23 @@ Return optimal transport coupling `γ` of the same dimensions as `C` which solve
 
 where ``H`` is the entropic regulariser, ``H(\\gamma) = -\\sum_{i, j} \\gamma_{ij} \\log(\\gamma_{ij})``.
 
+Note that for a common cost `C`, multiple histograms may be provided for a batch computation by passing `mu` and `nu`
+as matrices whose columns `mu[:, i]` and `nu[:, i]` correspond to pairs of histograms. 
+The output in this case is a vector `γ` of coupling matrices such that `γ[i]` is a coupling of `mu[:, i]` and `nu[:, i]`.
+
 """
 function sinkhorn(mu, nu, C, eps; kwargs...)
     # compute Gibbs kernel
     K = @. exp(-C / eps)
-
     # compute dual potentials
     u, v = sinkhorn_gibbs(mu, nu, K; kwargs...)
-
-    return Diagonal(u) * K * Diagonal(v)
+    if (u isa Vector) && (v isa Vector)
+        return Diagonal(vec(u)) * K * Diagonal(vec(v))
+    elseif (u isa Matrix) && (v isa Matrix)
+        return [Diagonal(u[:, i]) * K * Diagonal(v[:, i]) for i in 1:size(mu, 2)]
+    else
+        throw(ArgumentError("Error: sinkhorn_gibbs returned inconsistent outputs"))
+    end
```

Comment:
I have a different suggestion that would also add support for arbitrary arrays of histograms that can be broadcasted together (e.g., one source histogram and multiple target histograms and vice versa). The suggestion would be to write
```suggestion
    plan = K .* add_singleton(u, Val(2)) .* add_singleton(v, Val(1))
    return plan
```
and to define
```suggestion
function add_singleton(x::AbstractArray, ::Val{dim}) where {dim}
    shape = ntuple(ndims(x) + 1) do i
        return i < dim ? size(x, i) : (i > dim ? size(x, i - 1) : 1)
    end
    return reshape(x, shape)
end
```
Then with one source and one target histogram of length `m` and `n` a plan of size `(m, n)` would be returned. If there a `k` source histograms (size `(m, k)`), `k` plans are returned as array of size `(m, n, k)`. If there are `l` target histograms (size `(n, l)`), `l` plans are returned as array of size `(m, n, l)`. If there are `k` source and target histograms (size `(m, k)` and `(n, k)`), `k` pairwise plans are returned as array of size `(m, n, k)`. And if there are `k` source histograms (size `(m, k)`) and `l` target histograms (size `(n, 1, l)`), then `k x l` plans are returned as array of size `(m, n, k, l)`.

Reply:
Ok I'll give this a try. It certainly seems more natural than returning either a Matrix or Vector{Matrix}. 

One issue though, I don't think this will work in the setting where there are a different number of source and target histograms, since AFAIK the dual variables can't be mixed and matched between different OT instances. I.e. to compute OT from `p` to `[q1, ..., qn]` still requires `2n` dual variables.

Reply:
Ok I've adopted this change. `sinkhorn2` now returns in all cases an `Array` (never a scalar), for the sake of type stability. The output is the same as that of `PythonOT.sinkhorn2`.